src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/DecimalMathFuncs.txt

```diff
@@ -0,0 +1,453 @@
+
+>> Int( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(100000.12)] )
+Table({Value:1000000000000000000000001},{Value:2000000000000000000000002},{Value:100000})
+
+>> Trunc( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(100000.12)] )
+Table({Value:1000000000000000000000001},{Value:2000000000000000000000002},{Value:100000})
+
+>> Round( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.12})
+
+>> RoundUp( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.13})
+
+>> RoundDown( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(30000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:30000.12})
+
+// these rounding tests are returning the correct results, due to the conversion between decimal and float
+
+>> Int( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Trunc( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Round( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundUp( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.13},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundDown( [Float(30000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:30000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+// non-decimal aware numeric functions
+
+>> Sqrt( 1524157899404237160815376 )         // exact square of 1234567899876, should be done in float and match next test
+1234567899876
+
+>> Sqrt( Float(1524157899404237160815376) )  // exact square of 1234567899876
+1234567899876
+
+>> Sqrt( 4.000000000000000004 )              // outside of float range, will round to 4
+2
+
+>> Sqrt( 4.0000000004 )                      // within float range
+2.0000000001
+
+>> Log( 100, 10 )
+2
+
+>> Log( 10000000000000050000000, 10 )         // input will be rounded with float conversion
+22
+
+>> Log( 10000000000000050000000, Float(10) )
+22
+
+>> Log( 10000000000000000000000, 10 )         // above tests should match these answers
+22
+
+>> Log( 10000000000000000000000, Float(10) )  
+22
+
+>> Log( Float("1e100"), 10 )
+100
+
+>> Ln( 2.71828182845904523536028747135266249775724709369995 )
+1
+
+>> Ln( 100000000 )
+18.420680743952367
+
+>> Exp( 18.420680743952367 )
+100000000.00000007
+
+>> Pi()
```

Comment:
>Pi() [](http://example.com/codeflow?start=3&length=4)

We could return 3.1415926535897932384626433833 (or ...34 as next digit is 8)

Reply:
We could but then it would change the typing to decimal and would be inconsistent with the return type from Sin, Cos, etc.  If someone wants Pi to many digits, they can easily put it in a decimal themeslf.